src/model/gateway.rs

```diff
@@ -150,10 +154,12 @@ impl Activity {
     ///     Ok(())
     /// }
     /// ```
-    pub fn streaming<N, U>(name: N, url: U) -> Activity
+    pub fn streaming<'a, N, U>(name: N, url: U) -> Activity
     where
         N: ToString,
-        U: ToString,
+        Url: TryFrom<U>,
+        <Url as TryFrom<U>>::Error: Debug,
+        U: 'a,
```

Comment:
I cannot comprehend the need for the lifetime here. We do not express a relationship between types or references, therefore this lifetime is a no-op and unnecessary.
```suggestion
```